server/enclave/enc.cpp

```diff
@@ -0,0 +1,99 @@
+// Include the trusted modelaggregator header that is generated
+// during the build. This file is generated by calling the
+// sdk tool oeedger8r against the modelaggregator.edl file.
+#include "modelaggregator_t.h"
+
+#include <stdio.h>
+#include <vector>
+#include <numeric>
+#include <map>
+#include <set>
+#include <string>
+
+// Include encryption/decryption and serialization/deserialization headers
+#include "../../common/encryption/encrypt.h"
+#include "../../common/encryption/serialization.h"
+
+using namespace std;
+
+// This is the function that the host calls. It performs
+// the aggregation and encrypts the new model to pass back
+void enclave_modelaggregator(unsigned char*** encrypted_accumulator,
+            uint32_t* decrypted_accumulator_lengths,
```

Comment:
I think prefixing the length variables with `decrypted_` is potentially misleading since their values were never encrypted to begin with. I think it's best to drop this prefix.